crates/kubelet/src/device_plugin_manager/pod_devices.rs

```diff
@@ -0,0 +1,157 @@
+use super::DeviceIdMap;
+use crate::device_plugin_api::v1beta1::ContainerAllocateResponse;
+use k8s_openapi::api::core::v1::Pod;
+use kube::api::{Api, ListParams};
+use std::collections::{HashMap, HashSet};
+use std::sync::{Arc, Mutex};
+
+/// DeviceAllocateInfo contains the device ids reserved to a container for a specific
+/// resource and the ContainerAllocateResponse which contains information about what
+/// to mount into the container
+#[derive(Clone)]
+pub struct DeviceAllocateInfo {
+    /// device_ids contains the device Ids allocated to this container for the given resource name
+    pub device_ids: HashSet<String>, // ContainerAllocateRequest.devices_i_ds
+    pub allocate_response: ContainerAllocateResponse,
+}
+/// Map of devices allocated to the container, keyed by resource name
+pub type ResourceAllocateInfo = HashMap<String, DeviceAllocateInfo>;
+/// Map of container device information, keyed by container name
+pub type ContainerDevices = HashMap<String, ResourceAllocateInfo>;
+
+/// PodDevices contains the map of Pods to allocated devices
+#[derive(Clone)]
+pub struct PodDevices {
+    /// Map of devices allocated to the Pod keyed by Pod UID
+    allocated_devices: Arc<Mutex<HashMap<String, ContainerDevices>>>,
+    client: kube::Client,
+}
+
+impl PodDevices {
+    pub fn new(client: kube::Client) -> Self {
+        PodDevices {
+            allocated_devices: Arc::new(Mutex::new(HashMap::new())),
+            client,
+        }
+    }
+
+    /// get_active_pods activePods is a method for listing active pods on the node so the
+    /// amount of device plugin resources requested by existing pods could be counted
+    /// when updating allocated devices
+    pub async fn get_active_pods(&self) -> anyhow::Result<HashSet<String>> {
+        // TODO: should this be namespaced?
+        let pod_client: Api<Pod> = Api::all(self.client.clone());
+        let pods = pod_client.list(&ListParams::default()).await?;
```

Comment:
This list params should be limiting the pods by node name

Reply:
Thanks! This will definitely reduce the load. Fix: b1f3a819d19